GTFO-API/Utilities/Impl/ThreadDispatcher_Impl.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using AssetShards;
+using UnhollowerBaseLib.Attributes;
+using UnityEngine;
+
+namespace GTFO.API.Utilities.Impl
+{
+    internal class ThreadDispatcher_Impl : MonoBehaviour
+    {
+        public ThreadDispatcher_Impl(IntPtr intPtr) : base(intPtr) { }
+
+        static ThreadDispatcher_Impl()
+        {
+            AssetShardManager.add_OnStartupAssetsLoaded((Action)OnAssetsLoaded);
```

Comment:
Use AssetAPI's event instead
```suggestion
            AssetAPI.OnStartupAssetsLoaded += OnAssetsLoaded;
```